presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -503,6 +503,30 @@ protected Scope visitRenameSchema(RenameSchema node, Optional<Scope> scope)
         @Override
         protected Scope visitCreateTable(CreateTable node, Optional<Scope> scope)
         {
+            validateTableProperties(node.getProperties(), scope);
+            return createAndAssignScope(node, scope);
+        }
+
+        private void validateTableProperties(List<TableProperty> properties, Optional<Scope> scope)
+        {
+            Set<String> propertyNames = new HashSet<>();
+            for (TableProperty property : properties) {
+                if (!propertyNames.add(property.getName().getValue())) {
+                    throw new SemanticException(DUPLICATE_SCHEMA_PROPERTY, property, "Duplicate table property '%s'", property.getName().getValue());
```

Comment:
This is the wrong error code for table properties. Probably the easiest thing is to pass in the error code to use on failure.